docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,201 @@
+
+## Use Cases
+
+1. Migrating existing architecture from another platform (CFN, CDK, Terraform, etc.)
+    1. Resources exist but have never been managed by K8s - no spec files
+3. Migrating architecture across K8s environments
+    1. Resource are being managed by another K8s environment
+
+# Proposed Implementation
+
+## Overview
+
+The proposed solution will provide an alternate workflow by which to adopt resources through a separate set of manifests. A new custom resource definition, the `AdoptedResource` CRD, provides a signal to ACK to adopt an existing resource. 
+The `AdoptedResource` CRD references an existing AWS resource (either through ARN, name or other unique identifier) and a target Kubernetes `GroupVersionKind`. Each ACK controller will reconcile all adopted resources that match any of the types it controls (for the service it manages).
+The manager will describe the referenced AWS resource, typically through a `Describe*` SDK call, and use the returned values to compile a new custom resource - with filled spec and status fields. The manager will then apply this custom resource to the cluster using the metadata applied in the `AdoptedResource` spec.
+
+## `AdoptedResource` Specification
+
+The `AdoptedResource` custom resource definition will allow for specification of an AWS resource unique identifier (in the `aws` field) and a target `kubernetes` field - containing Kubernetes `GroupVersionKind` and `metadata` sub-fields.
+
+The `aws` resource unique identifier field allows identification through at most one of:
+
+* `arn` - An [ARN](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (Amazon Resource Name)
+* `nameOrID` - A resource name or identifier (a string value)
+
+The custom resource definition makes no assumptions about which fields should be present for any given target kind. It is therefore up to the manager to validate the identifier input shape for existence and correctness.
+
+The `kubernetes` field wraps the structures from `GroupVersionKind`, which specify the target ACK type through the following required fields:
+
+* [API group and version](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups-and-versioning)
+* [Resource kind](https://kubernetes.io/docs/reference/kubectl/overview/#resource-types)
+
+A validating webhook for the `AdoptedResource` custom resources will validate the existence of the target type in the cluster and ensure the requested type is able to be adopted. 
+
+The `kubernetes.metadata` field is an optional [`ObjectMeta`](https://github.com/kubernetes/apimachinery/blob/4f505736214f6a32e70f37c6bae217360fe227b2/pkg/apis/meta/v1/types.go#L110) type that will be applied as the metadata for the created custom resource. If the `name` and/o `namespace` fields within the `metadata` spec is not defined, the respective fields from the `AdoptedResource` will be used as the value for the custom resource.
```

Comment:
Technically, it's a `PartialObjectMeta`, not an `ObjectMeta`, right?

Reply:
Yes. Implementation detail, or worth putting in the proposal? Functionally identical to the end user

Reply:
nope, just wanted to make sure I wasn't crazy :)